Fix recent queries not working for DQL and Lucene

ahopp opened the pull request:
## Description
Fixes the Recent Queries button not working for DQL and Lucene query languages in the Discover plugin.

## Issues Resolved
Fixes #10810

## Root Cause
The Recent Queries feature was broken for DQL (kuery) and Lucene queries due to two issues in the query editor (`src/plugins/data/public/ui/query_editor/query_editor.tsx`):

**Visibility condition bug**: The `RecentQueriesTable` component visibility was gated by `isRecentQueryVisible && useQueryEditor`. The `useQueryEditor` flag is `false` for DQL and Lucene queries, preventing the dropdown from ever showing.

**Missing history save**: The `onSubmit` function never called `queryString.addToQueryHistory()`, so queries were not being saved to history in the first place.

## Changelog

##### `src/plugins/data/public/ui/query_editor/query_editor.tsx`

**Line 487**: Removed `&& useQueryEditor` from `RecentQueriesTable` visibility condition
- Before: `isVisible={isRecentQueryVisible && useQueryEditor}`
- After: `isVisible={isRecentQueryVisible}`

**Lines 157-160**: Added `queryString.addToQueryHistory()` call in `onSubmit` function

>      // Add query to queryString history for Recent Queries feature
>      if (currentQuery.query?.trim()) {
>        queryString.addToQueryHistory(currentQuery, dateRange);
>      }

## Testing
Tested locally with DQL and Lucene queries in the Discover plugin:
  - Executed multiple DQL and Lucene queries (e.g., order_id:570663, day_of_week:Thursday)
  - Clicked the "Recent queries" button
  - Verified that the dropdown appears and shows all executed queries
  - Confirmed queries can be re-run by clicking them from the list

DQL:
<img width="2272" height="336" alt="Image" src="https://github.com/user-attachments/assets/a58ce6ef-f5e7-4e69-9844-0bb3838906f1" />

Lucene:
<img width="2240" height="296" alt="Image" src="https://github.com/user-attachments/assets/37faa2b0-5bb2-4fc3-9d53-db54ede15014" />

### Check List
  - All tests pass
  - New functionality includes testing
  - Commits are signed per the DCO using --signoff

ahopp committed:
Fix recent queries not working for DQL and Lucene

The Recent Queries button was not working for DQL and Lucene query
languages due to two issues in the query editor:

1. The RecentQueriesTable component visibility was gated by the
   useQueryEditor flag, which is false for DQL/Lucene queries.

2. The onSubmit function never called addToQueryHistory(), so
   queries were not being saved to history in the first place.

This fix removes the useQueryEditor condition from the table visibility
and adds the addToQueryHistory() call in onSubmit to ensure all query
languages are properly saved and displayed in recent queries.

Fixes #10810

Signed-off-by: Andrew Hopp <andrew.hopp@me.com>

ahopp committed:
Add changeset file for PR #10837

Signed-off-by: Andrew Hopp <andrew.hopp@me.com>

ruanyl committed:
Merge branch 'main' into debug-recent-queries-10810

ahopp committed:
Merge branch 'main' into debug-recent-queries-10810

ahopp committed:
Merge branch 'main' into debug-recent-queries-10810

ahopp committed:
Merge branch 'main' into debug-recent-queries-10810

angle943 committed:
Merge branch 'main' into debug-recent-queries-10810

LDrago27 commented:
@ahopp Thanks for the change. 

Currently for the PPL, SQL languages the recent queries exists as a button on the footer of the query editor. For Lucene/DQL we are using a compressed version of the query editor which lacks the footer. Your change introduces the table but there isn;t a way to show/hide the recent queries table. 

Current Behaviour for PPL/SQL
<img width="1724" height="963" alt="Screenshot 2025-11-19 at 12 46 16 PM" src="https://github.com/user-attachments/assets/deea5316-8362-48f9-a996-c1d9a95731e2" />

@angle943 @ruchidh any suggestion on how do we handle this before introducing this change.

LDrago27 reviewed (dismissed):
Please refer to this comment https://github.com/opensearch-project/OpenSearch-Dashboards/pull/10837#issuecomment-3554574927

ahopp commented:
Actually, I think I'll also have to adjust the container styling as well. I think the current CX (non-persistent) allows for overlapping of the lower elements for DQL/Lucene while the PPL/SQL CX has a container with padding. I'll take a look, but let me know if that makes sense...and what you are asking for before we can fix this bug.

ahopp committed:
Make footer always visible for DQL and Lucene editors

- Remove conditional footer rendering based on focus state
- Restructure DOM to move footer outside form control wrapper
- Add wrapper div to properly contain editor and footer
- Update footer CSS to prevent content overlap
- Ensure consistent UX with PPL/SQL multi-line editors

ahopp committed:
Merge branch 'main' into debug-recent-queries-10810

LDrago27 reviewed (approved):
Looks Good. @ahopp

ruchidh committed:
Merge branch 'main' into debug-recent-queries-10810
